src/snowflake/cli/api/config.py

```diff
@@ -340,9 +340,18 @@ def _get_envs_for_path(*path) -> dict:
     }
 
 
-def _dump_config(conf_file_cache: Dict):
+def _dump_config(whole_conf_cache: Dict):
+    config_toml_dict = whole_conf_cache.copy()
+
+    if CONNECTIONS_FILE.exists():
+        # update connections in connections.toml
+        _update_connections_toml(whole_conf_cache.get("connections") or {})
```

Comment:
Won't it add all connections from config.toml to connections.toml?

Reply:
No, it won't. It will add only connections loaded to config manager and if CONNECTIONS_FILE exists then there will be no config.toml connections there (in the manager).

There is a test assertion for it (that "jwt" connection from config.toml isn't copied to connections.toml): https://github.com/snowflakedb/snowflake-cli/pull/1791/files#diff-da24b355349b53e4c8047034767d008e4c2ce9ea6c8933a3acd50d0828f00528R276

Reply:
Let's add this comment. Imho it's not obvious that only connection.toml connections are loaded when both files are present and ahve connection details

Reply:
Done :)